common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
@dsjen what are your thoughts on moving the `find_courses` function to a more common location?  It's very relevant to the changes in this PR but the import dependency from common/djangoapps/enrollment to lms/djangoapps/oauth2_handler is problematic.